src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutMissingTagCheck.java

```diff
@@ -151,8 +165,16 @@ private boolean intersectingWithMoreThan(final Edge edge)
                         .filter(HighwayTag::isCarNavigableHighway)
                         .filter(obj -> obj.getTag(ServiceTag.KEY).isEmpty())
                         // de-duplication sectioned edges
-                        .forEach(wayId -> connectedEdges.add(wayId.getOsmIdentifier())));
-        return connectedEdges.size() > MINIMUM_INTERSECTION;
+                        .forEach(wayId ->
+                        {
+                            if (wayId.getOsmIdentifier() == edge.getOsmIdentifier())
+                            {
+                                return;
+                            }
+                            connectedEdges.add(wayId.getOsmIdentifier());
```

Comment:
I would reverse the conditional here and make the connectedEdges.add inside the conditional and remove the "return".
```suggestion
                            if (wayId.getOsmIdentifier() != edge.getOsmIdentifier())
                            {
                                connectedEdges.add(wayId.getOsmIdentifier());
                            }
```